src/useAddonState/server.ts

```diff
@@ -0,0 +1,88 @@
+import type { Channel } from "@storybook/channels";
+
+import { getValue, GetValuePayload, setValue, SetValuePayload } from "./common";
+
+class AddonState {
+  private values: Record<string, any> = {};
+
+  private listeners: Record<string, ((value: any) => unknown)[]> = {};
+
+  get(key: string) {
+    return this.values[key];
+  }
+
+  set(key: string, value: any) {
+    this.values[key] = value;
+    (this.listeners[key] || []).forEach((l) => l(value));
+  }
+
+  listen<T>(key: string, cb: (value: T) => unknown) {
+    this.listeners[key] = [...(this.listeners[key] || []), cb];
+  }
+
+  // for testing
+  reset() {
+    this.values = {};
+    this.listeners = {};
+  }
+}
+
+const addonState = new AddonState();
```

Comment:
Why use a class at all? I don't see how it's beneficial.

Reply:
I guess no particular reason. It lets you have private members?